pkg/totalaccountwatcher/totalaccountwatcher.go

```diff
@@ -48,6 +48,7 @@ func Initialize(client client.Client, watchInterval time.Duration) {
 	}
 
 	TotalAccountWatcher = NewTotalAccountWatcher(client, AwsClient, watchInterval)
+	TotalAccountWatcher.Total = 99999 // Init this with a stupid high value to account for multiple aws calls on multiple account watchers starting at the same time.
```

Comment:
Yeah, it's definitely a low-LOE fix.

I recently proposed making the TotalAccountWatcher have the methodology which gates account creation based on the number of accounts.

Basic Idea: We have a new var inside of the TotalAccountWatcher called (naming is hard so this is just the idea) `TotalAccountWatcher.AccountsCanBeCreated` which is a bool.  Then account_controller doesn't do any of the logic to get the total number of accounts and compare it to the limit, that's handled within account_controller and that bool is updated.  That bool would default to false and accounts cannot be created until the initial run of the AccountWatcher is completed.

There would also need to be an update hook here so that we could tell the Total AccountWatcher that we've created an account so it can temporarily update the amount of accounts for the edge-cases when we're nearing the limit of accounts we have set.

Reply:
@2uasimojo yeah, makes sense, it's a cleaner approach and quick to implement

Reply:
I have to check for a zero-value here because int cannot be compared to nil, but we just have to keep this in mind if we ever run this on a fresh account, but I'm not even sure that would come in at 0 because it would at least have itself in the account?  Not sure, but we should just keep it in the back of our collective mind in case it ever comes up.

Reply:
Not comparing the total to `nil`, comparing the whole object to `nil`. Let's go over this tomorrow.